src/TurnerSoftware.SitemapTools/SitemapQuery.cs

```diff
@@ -178,12 +179,12 @@ public async Task<SitemapFile> GetSitemapAsync(Uri sitemapUrl, CancellationToken
 								}
 							}
 						}
-						else
+						else if (!permitInvalidOperations)
```

Comment:
This one I wouldn't bother putting the if-check on as to get here, we have to already accept the content type so our lookup here is more a formality as (currently) you can modify the `SitemapParsers` static variable directly. Like, for this to throw means either there is a fault with the library (a parser wasn't registered for a sitemap type) or someone messed with the static variable.

Reply:
In the future, the whole checking of sitemap types etc could be simplified a bit - right now it is a little cumbersome.

Reply:
How can a user modify the static `SitemapParsers` variable?
I looked into the feasibility of registering my own handler, but am unaware of a way to override a static member like that.

I was considering a good expansion would be to have a way that a user can register a new handler, but then the `SitmapParsers` can't be static.

Reply:
It is a [public static dictionary](https://github.com/TurnerSoftware/SitemapTools/blob/main/src/TurnerSoftware.SitemapTools/SitemapQuery.cs#L24) - basically at any point in your application you _could_ add or replace an entry. Similarly the `SitemapTypeMapping` dictionary too. With both, the only limiting factor is that the dictionaries map to an enum which you can't change but really, it is really either XML or Text that someone would be dealing with.

Admittedly since originally writing this code, I very much dislike exposing things like that. The flexibility isn't really useful for packages like this where the functionality is pretty simple and would rarely benefit people. If I was going to do an overhaul of the library, that would be one of the first things to go.